internal/app/machined/pkg/adapters/network/bridge_master_spec.go

```diff
@@ -0,0 +1,63 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package network
+
+import (
+	"github.com/mdlayher/netlink"
+	"golang.org/x/sys/unix"
+
+	"github.com/talos-systems/talos/pkg/machinery/resources/network"
+)
+
+// BridgeMasterSpec adapter provides encoding/decoding to netlink structures.
+//
+//nolint:revive,golint
+func BridgeMasterSpec(r *network.BridgeMasterSpec) bridgeMaster {
+	return bridgeMaster{
+		BridgeMasterSpec: r,
+	}
+}
+
+type bridgeMaster struct {
```

Comment:
Any reason to not make it public and make `revive` linter happy here?

Reply:
Just did it. There wasn't any reason other than it being c/p from `bondSpec` :). Don't know what was the reason for @smira to do it that way on bonds.

Reply:
I did it this way to have a better looking API: I don't need to make this type public, as it has no use outside of this package. The only public API is the function and the value it returns has a couple of methods.

Linters sometimes don't see the full picture. Yes, it's weird in general to return private types, but in this specific case it makes sense, as it's an adapter which extends a resource with additional methods.